### PR TITLE
Add no-dataflux option to text-based Lightning demo

### PR DESCRIPTION
**Pull request by MattIrv**
- [x] Tests pass
- [x] Appropriate changes to documentation are included in the PR

### CODE REVIEW

**Review by jdnurme — APPROVED** [review]

File: demo/lightning/text-based/single-node/model.py [hunk 1]
```diff
@@ -134,17 +136,29 @@ def format_data(raw_bytes):
...
         distribution of a parquet file, where batch-size corresponds to rows of the parquet table.
         """
 
+    def fsspec_iter(self, objects):
+        """
+        Return a generator that reads the given objects using fsspec instead of Dataflux.
+        """
+        for obj in objects:
+            object_name = obj[0]
+            path = f"gs://{self.bucket_name}/{object_name}"
```
Comment (jdnurme) [inline demo/lightning/text-based/single-node/model.py:157]: I'm assuming this just works with background auth?

**Review by jdnurme — APPROVED** [review]

**Review by Yash9060 — APPROVED** [review]

File: demo/lightning/text-based/distributed/model.py [hunk 2]
```diff
@@ -206,15 +221,20 @@ def __iter__(self):
         end_point = start_point + files_per_node
         if worker_info is None:
             print("Single-process data loading detected", flush=True)
-            for bytes_content in df_iter.dataflux_core.download.dataflux_download_lazy(
+            if self.no_dataflux:
```
Comment (awonak) [inline demo/lightning/text-based/distributed/model.py:224]: The `__iter__` method has quite a bit of conditional branching logic. It's a relatively simple class, so I would suggest considering separating this into two separate classes (e.g. `ParquetDatafluxIterableDataset` and `ParquetFsspecIterableDataset`) and using the flag to decide which dataset class to construct in the main method. Same suggestion for the single node demo too.

**Review by awonak — APPROVED** [review]

File: demo/lightning/text-based/distributed/model.py (hunk 2 above)
Comment (jdnurme) [reply demo/lightning/text-based/distributed/model.py:224]: We do have tasks slated to make this a standalone dataset in the future, which will probably incorporate the changes you're requesting here.

File: demo/lightning/text-based/distributed/mirvine-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  name: mirvine-1
+  labels:
+    kueue.x-k8s.io/queue-name: multislice-queue  # Name of the LocalQueue
+  annotations:
+    alpha.jobset.sigs.k8s.io/exclusive-topology: cloud.google.com/gke-nodepool # 1:1 job replica to node pool assignment
+spec:
+  failurePolicy:
+    maxRestarts: 0
+  replicatedJobs:
+    - name: mirvine-job-1
```
Comment (awonak) [inline demo/lightning/text-based/distributed/mirvine-deploy.yaml:27]: Should we replace user specific values with placeholders like "`<job-name>`"?

Comment (MattIrv) [reply demo/lightning/text-based/distributed/mirvine-deploy.yaml:27]: Will remove this prior to submitting, this file is just here so I don't have to edit the existing one that has placeholders

File: demo/lightning/text-based/single-node/model.py (hunk 1 above)
Comment (MattIrv) [reply demo/lightning/text-based/single-node/model.py:157]: Yes

**Comment by MattIrv** [comment]
Added the image segmentation code in too. @abhibyreddi and @jdnurme can you take another look?

**Review by abhibyreddi — APPROVED** [review]
Everything looks good on the image segmentation side.

**Pull request closed without merging**